File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/1440-convert-integer-to-the-sum-of-two-no-zero-integers.cpp
class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        int a,b;
        for(int i=1;i<n+1;i++){
            a=i,b=n-i;
            if (!brr(a) && !brr(b)){
                return {a,b};
            }   
    }
    return {};
}
    bool brr(int k){
        while(k>0){
            if(k%10==0){
                return true;
            }
            k/=10;
        }
        return false;
    }
};

File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/1440-convert-integer-to-the-sum-of-two-no-zero-integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        # a=b=0
        # for i in range(1,n+1):
        #     a=i
        #     b=n-i
        #     if str(a).find('0')==-1 and str(b).find('0')==-1:
        #         return [a,b]
        a,b=n,0
        pv=1
        while n>1:
            t=1
            if n%10==1: t=2
            a=a-t*pv
            b=b+t*pv
            n=(n-t)//10
            pv*=10
        return [a,b]